### PR TITLE
parser: support for new NEXT VALUES

### PR DESCRIPTION
**Pull request by sougou**
The new construct allows you to use the `SELECT NEXT N VALUES FROM t`. Consequently, the increment value in the sequence table is deprecated (and ignored). We still support the previous
`SELECT NEXT VALUE...` construct, which is now essentially `SELECT NEXT 1 VALUES...`. Once the new construct is adopted, we'll deprecate the old one.
The returned value is the first of the N values.

### CODE REVIEW

**Comment by alainjobart** [comment]
Why did you remove the increment? The returned values don't need to be sequencial, do they? Oh wait, you're returning only the first value, and reserving N values? Why not return all values?

**Comment by sougou** [comment]
There are a few reasons for this approach:
* The need for a static increment in the schema comes from people wanting to manage their own non-overlapping auto-incs for each shard. Because we provide a globally unique auto-inc, the general feedback from users is that they don't need this flexibility.
* The `NEXT N VALUES` construct functionally overlaps with the static increment because it amounts to specifying the increment on-the-fly. In other words, one can achieve everything they want with the new construct. There's really no use case left for a static increment.
* I return the first value mainly to be in line with how MySQL returns the insert id for multi-value inserts. We could return all, but it's redundant.

File: go/vt/tabletserver/query_executor.go [hunk 1]
```diff
@@ -332,29 +340,27 @@ func (qre *QueryExecutor) execNextval() (*sqltypes.Result, error) {
...
-			inc, err := qr.Rows[0][2].ParseInt64()
-			if err != nil {
-				return nil, fmt.Errorf("error loading sequence %s: %v", qre.plan.TableName, err)
-			}
-			if inc < 1 {
-				return nil, fmt.Errorf("invalid increment for sequence %s: %d", qre.plan.TableName, inc)
+			newLast := nextID + cache
+			for newLast <= t.NextVal+inc {
+				newLast += cache
 			}
```
Comment (harshit-gangal) [inline go/vt/tabletserver/query_executor.go]: Instead of loop we can have something like this
newLast := nextID + ceil(inc/cache) * cache`

Comment (sougou) [inline go/vt/tabletserver/query_executor.go]: Based on offline discussion, there's a risk to using ceil because of possible off-by-epsilon errors with floating point numbers.
It's probably feasible to come up with an expression using integer-only arithmetic. However, the loop is easy to prove correctness, and is something that will rarely get executed because cache will typically be bigger than inc.
TL;DR: We'll leave it as is for now.

**Comment by harshit-gangal** [comment]
LGTM

**Comment by alainjobart** [comment]
LGTM
